### PR TITLE
Enabled and configured the Scala coverage plugin to fail builds if (for discussion).

### PR DESCRIPTION
**Pull request by alexec**
…0% unit + functional coverage occurred.

This is for discussion. Would test coverage be useful? Have you found it identifies problems in the past? What happens when it falls below a value? Drop the value, address the issue?

### CODE REVIEW

**Comment by danmassie** [comment]
I personally don't find test coverage metrics useful. In my experience having that certain value to aim for leads to poor tests being written just to meet the standard.

The testing approach we have opted for so far is a functional approach for most cases and unit testing for discrete components such as validation and calculations.

**Comment by alexec** [comment]
Interesting. I've seen code coverage been useless on some code bases (sizeable untested, analysis run nightly by CI). I've also seen it prevent bugs going live in others (domain logic problems, diligent teams running analysis ad-hoc pre-commit). Current coverage on this project is 92%, which shows that there has been good test coverage without prompting. 

I think it would become more useful when we start implementing the calculator.

I've found mutation coverage much more useful (regularly finding real bugs), but Scala does not have any tooling for mutation testing.

I'd be interested to see feedback from the rest of the team.

**Comment by danmassie** [comment]
+1 on mutation testing.

File: project/plugins.sbt [hunk 1]
```diff
@@ -9,3 +9,5 @@ addSbtPlugin("uk.gov.hmrc" % "sbt-git-versioning" % "0.8.0")
 addSbtPlugin("uk.gov.hmrc" % "sbt-distributables" % "0.10.0")
 
 addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.3.10")
+
+addSbtPlugin("org.scoverage" % "sbt-scoverage" % "1.3.5")
```
Comment (dabd) [inline project/plugins.sbt]: test comment

**Comment by saurabharora80** [comment]
No Thanks! :D

**Comment by danmassie** [comment]
Closing PR

**Pull request closed without merging**